### PR TITLE
openimageio: add version 2.4.7.1

### PR DESCRIPTION
**Pull request by Latios96**
Specify library name and version:  **openimageio/2.4.7.1**

#### Patches
- Patch `2.4.7.1-cmake-targets.patch`: Ensure the project builds correctly with Conan,.  The patch is similar to the patches from previous versions
- Patch `2.4.7.1-fix-msvc2017.patch`: Fixes compile error with MSVC 2017 (see [Issue at OpenImageIO](https://github.com/OpenImageIO/oiio/issues/3762)). I chose the patch_type `official`, cause the fix was merged upstream (see [PR at OpenImageIO](https://github.com/OpenImageIO/oiio/pull/3763)), but there is no version released with the fix yet.

#### Dependencies:
Updated dependencies to newer versions
##### OpenCV
Deactivated by default for now, because of conflicting dependencies of prebuild packages. OpenImageIO depends on OpenColorIO and OpenCV, which both depend on OpenEXR. OpenColorIO in Conan Center is build using openexr/2.5.7 and OpenCV packages are build using openexr/3.1.5. Because of that, OpenImageIO won't build in CI, because of missing packages.  

It would be possible to update OpenColorIO's version of OpenEXR, I think this will be done in #15669.

I decided to disable OpenCV for now. I think it's more relevant to have a recent version of OpenImageIO in Conan Center than having a prebuild binary with OpenCV.

##### onetbb
Deactivated because of missing prebuild binaries. It seems like there are no prebuild binaries for the `MTd` runtime, also not for newer versions of onetbb. I am not sure how to deal with that, I disabled onetbb by default for now, so CI passes.

#### Various
- fixed link error with TBB (in cpp_info `main` required TBB before, which was not correct, has to be `openimageio_util`)
- fixed linter errors

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by uilianries** [comment]
@Latios96 Please wait for now, we restarted the CI and your job is waiting in the queue. If you close/open again, it will be moved to the end of the queue automatically.

File: recipes/openimageio/all/conanfile.py [hunk 1]
```diff
@@ -129,38 +132,45 @@ def requirements(self):
...
+            **self.conan_data["sources"][self.version],
+            strip_root=True,
+            destination=self._source_subfolder
+        )
 
     def _patch_sources(self):
-         for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            files.patch(self, **patch)
```
Comment (prince-chrismc) [inline recipes/openimageio/all/conanfile.py:166]: In 1.50 there's a new apply_conandata_patches helper for this

**Review by prince-chrismc — DISMISSED** [review]
This looks really good, great work!

**Comment by Latios96** [comment]
@prince-chrismc thanks!

**Review by prince-chrismc — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

**Pull request merged**